packages/legend-graph/src/models/protocols/pure/v1/V1_PureGraphManager.ts

```diff
@@ -2016,6 +2036,237 @@ export class V1_PureGraphManager extends AbstractPureGraphManager {
     );
   }
 
+  async indexEntitiesWithDependencyIntoGraph<T extends V1_PackageableElement>(
```

Comment:
So, my thinking is we could potentially rename this method to build light graph with just indexing (similar to what you have here) but return a slightly more complicated object instead: separating the `graph` and the `dependencies` part. That way we could reuse for the future `buildLightGraph` method which we would expose at interface level, i.e. `AbstractPureGraphManager`.

Reply:
I'm not suggesting we have to do it now, but we should put a `TODO` if we don't.

Reply:
@akphi not sure i understand what you mean here. whats the purpose of returning this more complicated object ?